src/math/p5.Vector.js

```diff
@@ -1742,6 +1742,156 @@ p5.Vector = class {
     return this;
   }
 
+  /**
+ * Performs spherical linear interpolation with the other vector
+ * and returns the resulting vector.
+ * The result of slerping between 2D vectors is always a 2D vector.
+ *
+ * @method slerp
+ * @param {p5.Vector} v the p5.Vector to slerp to
+ * @param {Number} amt The amount of interpolation. some value between 0.0
+ *                     (old vector) and 1.0 (new vector). 0.9 is very near
+ *                     the new vector. 0.5 is halfway in between.
+ * @return {p5.Vector}
+ *
+ * @example
+ * <div class="norender">
+ * <code>
+ *
+ * const v1 = createVector(1, 0, 0);
+ * const v2 = createVector(0, 1, 0);
+ *
+ * const v = v1.slerp(v2, 1/3);
+ * print(v.toString());
+ * // v's components are almost [cos(30°), sin(30°), 0]
+ * </code>
+ * </div>
+ *
+ * <div>
+ * <code>
+ * let needle;
+ * function setup() {
+ *   createCanvas(100, 100);
+ *   stroke(0);
+ *   strokeWeight(4);
+ *
+ *   needle = createVector(50, 0);
+ * }
+ *
+ * function draw(){
+ *   background(255);
+ *   translate(50, 50);
+ *
+ *   const theta = Math.atan2(mouseY-50, mouseX-50);
+ *   const v = createVector(50 * Math.cos(theta), 50 * Math.sin(theta));
+ *   // slerp between v and needle
+ *   // needle vector is changed by slerp function.
+ *   needle.slerp(v, 0.05);
+ *
+ *   line(0, 0, needle.x, needle.y);
+ * }
+ * </code>
+ * </div>
+ *
+ * <div>
+ * <code>
+ * let v1, v2, v3;
+ * function setup(){
+ *   createCanvas(100, 100, WEBGL);
+ *   noStroke();
+ *   v1 = createVector(30, 0, 0);
+ *   v2 = createVector(0, 30, 0);
+ *   v3 = createVector(0, 0, 30);
+ * }
+ *
+ * function draw(){
+ *   background(0);
+ *   lights();
+ *   fill(255);
+ *   ambientMaterial(255);
+ *
+ *   const t = (frameCount % 60) / 60;
+ *   // v1, v2, v3 is not changed by slerp function.
+ *   // because this function is static version.
+ *   const v4 = p5.Vector.slerp(v1, v2, t);
+ *   const v5 = p5.Vector.slerp(v2, v3, t);
+ *   const v6 = p5.Vector.slerp(v3, v1, t);
+ *   translate(v4.x, v4.y, v4.z);
```

Comment:
For this example, I wonder if we can make what's happening a bit clearer:
- Since the spheres end in a configuration that looks identical to the start, it might accidentally look like the vectors keep moving, maybe coloring them differently and/or making them oscillate back and forth instead of jumping back to the start would make it easier to see what's happening?
- Although it's less lines of code, subtracting the previous point from the next might add a bit of confusion. Maybe we could push/pop even though it's longer? or use something like `line()` where we can use coordiantes without `translate()`?

As a quick test, do you think something like this is clearer? https://editor.p5js.org/davepagurek/sketches/NBexafWPU

```js

function draw(){
  background(255);

  const t = map(sin(frameCount/30), -1, 1, 0, 1);
  // v1, v2, v3 is not changed by slerp function.
  // because this function is static version.
  const v4 = p5.Vector.slerp(v1, v2, t);
  const v5 = p5.Vector.slerp(v2, v3, t);
  const v6 = p5.Vector.slerp(v3, v1, t);
  strokeWeight(10)
  strokeCap(SQUARE)
  stroke('red')
  line(0, 0, 0, v4.x, v4.y, v4.z);
  stroke('green')
  line(0, 0, 0, v5.x, v5.y, v5.z);
  stroke('blue')
  line(0, 0, 0, v6.x, v6.y, v6.z);
}
```

Reply:
I thought that example might be a little confusing for me too. It's true that "cylinders" or "lines" are easier to understand, so I'm going to rewrite them. I would like to adopt the "lines" because it is easier for the code to understand.

Reply:
How about this...?
[slerp sample for 3D](https://editor.p5js.org/dark_fox/sketches/iU8-el0jx)
```js
function setup(){
  createCanvas(100, 100, WEBGL);
}

function draw(){
  background(0);
  
  const vx = createVector(30, 0, 0);
  const vy = createVector(0, 30, 0);
  const vz = createVector(0, 0, 30);

  const t = map(sin(frameCount * TAU / 120), -1, 1, 0, 1);
  // v1, v2, v3 is not changed by slerp function.
  // because this function is static version.
  const vSlerpXY = p5.Vector.slerp(vx, vy, t);
  const vSlerpYZ = p5.Vector.slerp(vy, vz, t);
  const vSlerpZX = p5.Vector.slerp(vz, vx, t);
  strokeWeight(6);
  strokeCap(SQUARE);
  stroke("red");
  line(0, 0, 0, vSlerpXY.x, vSlerpXY.y, vSlerpXY.z);
  stroke("green");
  line(0, 0, 0, vSlerpYZ.x, vSlerpYZ.y, vSlerpYZ.z);
  stroke("blue");
  line(0, 0, 0, vSlerpZX.x, vSlerpZX.y, vSlerpZX.z);
}
```

Renamed variables to make it clearer which direction the axis is pointing.

Reply:
looks great!